mathics/eval/numbers.py

```diff
@@ -1,10 +1,109 @@
+from typing import Optional
+
+import mpmath
 import sympy
 
+from mathics.core.atoms import Complex, MachineReal, PrecisionReal, Real
 from mathics.core.convert.sympy import from_sympy
+from mathics.core.element import BaseElement
 from mathics.core.expression import Expression
+from mathics.core.number import MACHINE_PRECISION_VALUE, ZERO_MACHINE_ACCURACY, dps
 from mathics.core.symbols import SymbolPlus
 
 
+def eval_accuracy(z: BaseElement) -> Optional[float]:
+    """
+    Determine the accuracy of an expression expr.
+    If z is a Real value, returns the difference between
+    the number of significant decimal figures (Precision) and
+    log_10(z).
+
+    For example,
+    ```
+    12.345`2
+    ```
+    which is equivalent to 12.`2  has an accuracy of
+    ```
+    0.908509 == 2. -  log(10, 12.345)
+    ```
+
+    If the expression contains Real values, returns
+    the minimal accuracy of all the numbers in the expression.
+
+    Otherwise returns None, representing infinite accuracy.
+    """
+    if isinstance(z, MachineReal):
```

Comment:
Add type annotation for return type?

Reply:
where?

Reply:
My mistake - I don't know why I thought this  a function definition.